client/signup/steps/user/index.jsx

```diff
@@ -195,14 +195,38 @@ export class UserStep extends Component {
 
 		if ( [ 'wpcc', 'crowdsignal' ].includes( flowName ) && oauth2Client ) {
 			if ( isWooOAuth2Client( oauth2Client ) && wccomFrom ) {
-				subHeaderText =
-					'cart' === wccomFrom
-						? translate(
-								"You'll need an account to complete your purchase and manage your subscription"
-						  )
-						: translate(
-								"You'll need an account to connect your store and manage your extensions"
-						  );
+				switch ( wccomFrom ) {
+					case 'cart':
+						subHeaderText = translate(
+							"You'll need an account to complete your purchase and manage your subscription"
```

Comment:
I'm unsure how to test this flow 🤔

Reply:
I agree it doesn't seem to be used. When I click "Log in" from the cart page, `wccom-from` is empty, and on logging in I'm sent back to the cart page.

When I add `wccom-from%3Dcart` to the Calypso URL, I get this UI. (I suppose my logged-in state is overwriting the subheader text.) This is quite nice and there are probably more parts to it than this, so it could be OK to leave it for now, in case anyone comes round to this approach again. It seems various nice things have accidentally got lost from the flow over the years. cc @poligilad-auto for your interest.

<img width="500" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1647564/304e936b-e5fc-4232-99e8-ec252cb3c6a0">